web/profiles/custom/yalesites_profile/modules/custom/ys_node_access/src/EventSubscriber/NodeAccessEventSubscriber.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\ys_node_access\EventSubscriber;
+
+use Drupal\Core\EventSubscriber\HttpExceptionSubscriberBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Drupal\path_alias\AliasManager;
+
+/**
+ * Overrides the 403 response to perform a CAS redirect for specific pages.
+ */
+class NodeAccessEventSubscriber extends HttpExceptionSubscriberBase {
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The path alias manager.
+   *
+   * @var \Drupal\path_alias\AliasManager
+   */
+  protected $pathAlias;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The construct for this class.
+   *
+   * @param Drupal\Core\Session\AccountInterface $current_user
+   *   The current user service.
+   * @param Drupal\path_alias\AliasManager $path_alias
+   *   The path alias manager.
+   */
+  public function __construct(AccountInterface $current_user, AliasManager $path_alias) {
+    $this->currentUser = $current_user;
+    $this->pathAlias = $path_alias;
```

Comment:
Also, reading this again, noticed that the AliasManager services could be removed, you are not currently using it in your current logic

Reply:
Removed, thanks!